ejb/WFLY-8449_Return_context_data_to_client.adoc

```diff
@@ -0,0 +1,81 @@
+= Return Context Data to EJB Clients
+:author:            Stuart Douglas
+:email:             sdouglas@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This will allow for some data that has been placed in the InvocationContext by an EJB or Interceptor to
+be returned to the client so it can be processed by client interceptors. Only data that has explicitly
+been requested by the interceptor will be returned.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-8449[WFLY-8449]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-976[EAP7-976]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Affected Projects or Components
+
+ * Wildfly
+ * EJB Client
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+An `EJBClientInterceptor` will be able to request data that it is interested in by calling
+`org.jboss.ejb.client.EJBClientInvocationContext#addReturnedContextDataKey(String key)`. This tells the server that the
+client interceptor would like any data that is stored under the provided key in the context data map to be returned
+to the client.
+
+Internally this will create/add to a `Set` in the clients context data stored under
+`org.jboss.ejb.client.invocation.returned-context-data`, which will then be sent to the server as part of the invocations
```

Comment:
This key is pretty long.  Our keys start with just ``jboss.`` and the ``ejb.client.invocation`` is all inferrable (becuase it's an EJB invocation).  Maybe ``jboss.returned-keys``?

Reply:
makes sense

Reply:
As per client/server discussion below, maybe ``jboss.send-keys`` or something woudl be better.

Reply:
I don't really understand what you mean here

Reply:
Sending keys from client to server should work the same way as sending keys from server to client.  When sending from client to server, we're not "returning" the key, we're simply "sending" it.

The latter question is, do we always include the special key itself in the set of keys transmitted between client and server, or do we exclude it?